kernel/src/main/java/org/kframework/compile/checks/CheckSyntaxGroups.java

```diff
@@ -0,0 +1,46 @@
+package org.kframework.compile.checks;
+
+import org.kframework.definition.Sentence;
+import org.kframework.definition.SyntaxAssociativity;
+import org.kframework.definition.Tag;
+import org.kframework.utils.errorsystem.KEMException;
+import org.kframework.utils.errorsystem.KException;
+import org.kframework.utils.errorsystem.KExceptionManager;
+import org.kframework.definition.Module;
+import scala.collection.JavaConverters;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+
+public class CheckSyntaxGroups {
+
+    private final Set<KEMException> errors;
+    private final KExceptionManager kem;
+    private final Module module;
+
+    public CheckSyntaxGroups(Set<KEMException> errors, Module module, KExceptionManager kem) {
+        this.errors = errors;
+        this.kem = kem;
+        this.module = module;
+    }
+
+    public void check(Sentence s) {
+        if(s instanceof SyntaxAssociativity) {
+            Set<Tag> tags = JavaConverters.setAsJavaSet(((SyntaxAssociativity) s).tags());
```

Comment:
Usually we use `mutable` from `collections.scala`
You will need to add
```
import static org.kframework.Collections.*;
```
```suggestion
            Set<Tag> tags = mutable(((SyntaxAssociativity) s).tags());
```